pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -100,6 +102,20 @@ def __verify_eval_loop_configuration(self, model: 'pl.LightningModule', stage: s
             rank_zero_warn(f'you passed in a {loader_name} but have no {step_name}. Skipping {stage} loop')
         if has_step and not has_loader:
             rank_zero_warn(f'you defined a {step_name} but have no {loader_name}. Skipping {stage} loop')
+        if has_step and has_loader:
+            sig_args = signature(getattr(model, step_name)).parameters
+            loaders = getattr(model, loader_name)()
+            len_loaders = 1 if not isinstance(loaders, list) else len(loaders)
+            if (len(sig_args) == 2 and len_loaders != 1):
+                raise MisconfigurationException(
+                    f'You provide multiple {stage} dataloaders, but no `dataloader_idx` argument'
+                    f' given in {step_name}.'
+                )
+            if (len(sig_args) == 3 and len_loaders < 2):
+                raise MisconfigurationException(
+                    f'You provided only a single {stage} dataloader, but have included the `dataloader_idx`'
+                    f' in the {step_name} method.'
```

Comment:
what if someone has `dataloader_idx=0` in the signature, or just wants their module to be agnostic for as many dataloaders to validate as they want, including just one?

Reply:
fair point, in that case we should probably check if the user has provided a default value and if not then raise an error